#!/bin/sh /usr/share/dpatch/dpatch-run

## 13_directoryfix.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Improves OSD messages of currently processed recordings when the
## DP: recording is in a sub directory.
## DP: ( http://vdr-portal.de/board/thread.php?threadid=38968 )

@DPATCH@
diff -urNad vdr-addon-noad-0.7.0+update1~/svdrpc.cpp vdr-addon-noad-0.7.0+update1/svdrpc.cpp
--- vdr-addon-noad-0.7.0+update1~/svdrpc.cpp	2010-01-19 16:19:53.000000000 +0100
+++ vdr-addon-noad-0.7.0+update1/svdrpc.cpp	2010-01-28 23:27:06.000000000 +0100
@@ -256,13 +256,30 @@
 {
   char *baseName = NULL;
   char *cp = NULL;
-  const char *vend = strchr(filename,'/');
-  if( vend )
+  char *fname = NULL;
+  char *vend = NULL;
+  
+  asprintf(&fname, "%s", filename);
+  if(fname[strlen(fname) - 1] == '/')
+    fname[strlen(fname) - 1] = '\0';
+  vend = strrchr(fname,'/');
+  if(vend) {
+    *vend = '\0';
+    vend = strrchr(fname,'/');
+  }
+  
+  if( vend && vend[1] == '_')
+  {
+    *vend = '\0';
+    vend = strrchr(fname,'/');
+  }
+
+  if( vend && strchr(vend+1, '/'))
     vend = strchr(vend+1,'/');
   if( vend )
-    asprintf(&baseName,"mesg %s %s",msg,vend+1);
+    asprintf(&baseName,"mesg %s '%s'",msg,vend+1);
   else
-    asprintf(&baseName,"mesg %s %s",msg, filename);
+    asprintf(&baseName,"mesg %s '%s'",msg, filename);
   if( baseName[strlen(baseName)-1] == '/' )
     baseName[strlen(baseName)-1] = '\0';
   char *vend1 = strrchr(baseName, '/');
@@ -274,6 +291,7 @@
 
   free(baseName);
   free(cp);
+  free(fname);
 }
 
 void noadStartMessage( const char *s)
